### ISSUE TITLE
Proper Shoes::Widget support

### ISSUE THREAD

**Issue by noahgibbs** [scarpe-team/scarpe#342]
We had some trouble during Hack Days 2 because (very reasonably) people wanted to build some less-trivial chunks of UI, and we haven't implemented old school Shoes::Widget.

It's time.

This is going to require moving the current Shoes::Widget class to a different name -- we need the functionality, but that name means something else in Old Shoes. A Shoes::Widget is more of a DSL-builder for a Flow than a widget in the sense we currently mean. (Doc for flow assumption: Shoes-core has [widget inherit from flow](https://github.com/shoes/shoes4/blob/46fbd81a1f7e40736b48c600fd798fae5b79700d/shoes-core/lib/shoes/widget.rb#L30).)

Here's an example from examples:

```
class MenuPanel < Shoes::Widget
  @@boxes = []
  def initialize(color, args)
    @@boxes << self
    background color
    para link("Box #{@@boxes.length}", :stroke => white, :fill => nil).
      click { visit "/" },
        :margin => 18, :align => "center", :size => 20
    hover { expand }
  end
  def expand
    if self.width < 170
      a = animate 30 do
        @@boxes.each do |b|
          b.width -= 5 if b != self and b.width > 140
        end
        self.width += 5
        a.stop if self.width >= 170
      end
    end
  end
end

Shoes.app :width => 400, :height => 130 do
  style(Shoes::Link, :underline => nil)
  style(Shoes::LinkHover, :fill => nil, :underline => nil)
  menu_panel green,  :width => 170, :height => 120, :margin => 4
  menu_panel blue,   :width => 140, :height => 120, :margin => 4
  menu_panel red,    :width => 140, :height => 120, :margin => 4
  menu_panel purple, :width => 140, :height => 120, :margin => 4
end
```

Notice that the args when creating one (width, height, margin) are passed in to the Slot-type constructor. Those aren't explicitly used in initialize, they're used by the underlying stack-or-flow-or-whatever that Widget acts as.

Making it more annoying, this example isn't even calling super, so this doesn't work the obvious way. I think that's why Shoes4 deprecated this interface and renamed initialize as initialize_widget (https://github.com/shoes/shoes4/blob/main/shoes-core/lib/shoes/widget.rb), because implicitly calling a super without calling super is kinda Not Okay in Ruby, but Shoes only looks like Ruby, it's actually weird C extensions.

**Comment by noahgibbs** [comment]
PR #401 opens up Shoes::Widget in the namespace, after which this should be more possible.

**Comment by noahgibbs** [comment]
This is now the most immediate obstacle to running Hackety-Hack.

**Comment by noahgibbs** [comment]
Should be fixed by #414.

**Comment by noahgibbs** [comment]
Fixed now.
